### PR TITLE
API Cleaned up versioned status checks

### PR DESCRIPTION
**Pull request by tractorcow**
CMS PR to go with https://github.com/silverstripe/silverstripe-asset-admin/pull/284

Requires https://github.com/silverstripe/silverstripe-framework/pull/6136

### CODE REVIEW

File: code/Model/SiteTree.php [hunk 1]
```diff
@@ -2614,24 +2615,22 @@ public function setMenuTitle($value) {
...
-				} else {
-					$flags['archived'] = array(
-						'text' => _t('SiteTree.ARCHIVEDPAGESHORT', 'Archived'),
-						'title' => _t('SiteTree.ARCHIVEDPAGEHELP', 'Page is removed from draft and live'),
-					);
-				}
-			} else if($this->getIsAddedToStage()) {
+			if($this->isOnLiveOnly()) {
+				$flags['removedfromdraft'] = array(
+					'text' => _t('SiteTree.REMOVEDFROMDRAFTSHORT', 'Removed from draft'),
```
Comment (chillu) [inline code/Model/SiteTree.php]: Shouldn't the user-facing wording match the API change?

Comment (tractorcow) [inline code/Model/SiteTree.php]: Will set to `On live only`

Comment (tractorcow) [inline code/Model/SiteTree.php]: Done.

**Comment by tractorcow** [comment]
@ingo ready for re-review.

**Pull request merged**